core/defineStore.js

```diff
@@ -0,0 +1,19 @@
+import { listenKeys } from "nanostores";
+import { useCallback } from "preact/hooks";
+
+export function defineStore(store, options = {}, useSyncExternalStoreFn) {
```

Comment:
You are adding new methods like `defineStore`.

Can you explain why? Is it methods for end-user?

Reply:
This is an internal method of the package that the end user does not have access to.
This method contains the general logic for implementing the "useStore" and "useStoreSignal" hooks. As it became say earlier, these hooks have one implementation except for the "useSyncExternalStore" method.